source/extensions/http/cache/file_system_http_cache/cache_eviction_thread.cc

```diff
@@ -0,0 +1,79 @@
+#include "source/extensions/http/cache/file_system_http_cache/cache_eviction_thread.h"
+
+#include "envoy/thread/thread.h"
+
+#include "source/extensions/http/cache/file_system_http_cache/file_system_http_cache.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+namespace FileSystemHttpCache {
+
+CacheEvictionThread::CacheEvictionThread(Thread::ThreadFactory& thread_factory)
+    : thread_(thread_factory.createThread([this]() { work(); })) {}
```

Comment:
That's already the case - CacheSingleton is only allocated when getCache is called, and the CacheSingleton is deleted when the last cache goes out of scope, because the caches have shared_ptr ownership of the CacheSingleton.

Reply:
Added some log statements. Testing it on a live server is next in my set of things to do, but it'll be much easier to do that if this PR can be merged first. (We have an end-to-end test framework, but there's a whole dependency chain thing going on such that I'm only supposed to update to main-branch envoy.)